cmake/app.cmake

```diff
@@ -1,23 +1,26 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+# Create Mbed OS library
+add_library(mbed-os OBJECT)
+
+# Create application executable
+add_executable(app)
+
 include(.mbedbuild/mbed_config.cmake)
```

Comment:
Would it be worth moving these includes down into the `mbed-os` entry point? I think we should aim to eliminate `app.cmake` as it's currently coupled to both the library and the application. We should try and break this coupling so the application can just `target_link_libraries(mbed-os)` and everything works. I believe the added indirection of `app.cmake`, which seems to set properties that should belong in the `mbed-os` library, will be confusing to developers who have used CMake before.

Reply:
> Would it be worth moving these includes down into the mbed-os entry point? 

Can you elaborate ? 

app.cmake should look like after we do clean-up and fixes to linking :

```
include(.mbedbuild/mbed_config.cmake)
include(${MBED_ROOT}/cmake/toolchain.cmake)
include(${MBED_ROOT}/cmake/core.cmake)
include(${MBED_ROOT}/cmake/profile.cmake)
include(${MBED_ROOT}/cmake/env.cmake)
include(${MBED_ROOT}/cmake/util.cmake)

# Include Mbed OS main cmake
add_subdirectory(mbed-os)
```

This to move to main Mbed OS cmake so no need to have app.cmake ?

Reply:
Yes, I'm wondering if the `include` statements can be moved to `mbed-os/CMakeLists.txt` and in the top level application `CMakeLists.txt` we can just do something like:

```
add_executable(app)
add_subdirectory(mbed-os)
target_link_libraries(app mbed-os)
```

If we have a plan to do this in an incremental way, that's great. I'm just thinking about what the end result should look like.

Reply:
I will separate the change in two commits so the logical change is more transparent.

From the conversation above, I take we are not yet moving executable creation to the app CMake source file. There the PR will stay as is. Correct?

Reply:
I'm fine with changing it in another PR.

Reply:
agreed with @rwalton-arm about removing app.cmake